$OpenBSD: patch-i3bar_src_xcb_c,v 1.12 2014/01/11 16:04:43 dcoppa Exp $

commit 23b1da598ebac70d82070abb557bef0fd3fabce9
Author: Tony Crisci <tony@dubstepdish.com>
Date:   Thu Dec 26 00:30:21 2013 -0500

i3bar: Set `mapped` flag on trayclient creation

When a trayclient is first created as a structure in memory, explicitly
set the `mapped` flag to false. Otherwise it may initialize to `true` in
some circumstances without actually being mapped, causing a request to
be mapped from the client to be ignored.

Create the trayclient in memory before handling a request to be mapped
immediately.

--- i3bar/src/xcb.c.orig	Sun Dec 22 21:12:41 2013
+++ i3bar/src/xcb.c	Fri Jan 10 20:18:08 2014
@@ -603,17 +603,18 @@ static void handle_client_message(xcb_client_message_e
              * exits/crashes. */
             xcb_change_save_set(xcb_connection, XCB_SET_MODE_INSERT, client);
 
+            trayclient *tc = smalloc(sizeof(trayclient));
+            tc->win = client;
+            tc->xe_version = xe_version;
+            tc->mapped = false;
+            TAILQ_INSERT_TAIL(output->trayclients, tc, tailq);
+
             if (map_it) {
                 DLOG("Mapping dock client\n");
                 xcb_map_window(xcb_connection, client);
             } else {
                 DLOG("Not mapping dock client yet\n");
             }
-            trayclient *tc = smalloc(sizeof(trayclient));
-            tc->win = client;
-            tc->xe_version = xe_version;
-            TAILQ_INSERT_TAIL(output->trayclients, tc, tailq);
-
             /* Trigger an update to copy the statusline text to the appropriate
              * position */
             configure_trayclients();
